### ISSUE TITLE
Support composite repositories

### ISSUE THREAD

**Issue by lppedd** [JaneWardSandy/eclipse-pde-partial-idea#81]
Following issue #80, an enhancement to simplify Target Platform management could be supporting composite repositories.
https://wiki.eclipse.org/Equinox/p2/Composite_Repositories_(new)

Basically two files, `compositeContent.xml` and `compositeArtifacts.xml`, explain how to access the - real - children repositories.

**Comment by JaneWardSandy** [comment]
Yeah, `content` and `artifacts` seems identical, and those two file under `update.site` seems too.

Did we add all repository into target in one `Location` if we select the `composite repositories`? All repository's `bundle` and `feature` were merge into one `Location`, it will difficult to know the feature was under which repository.

if you don't care, I can add `composite repositories` support, and it can select the root directory, add all `feature` and `bundle` into one `Location` under children repositories.

**Comment by lppedd** [comment]
@JaneWardSandy I think the only two files you should be looking at are the ones in the root folder.

Reading them, you just "simulate" multiple directories being added, while in reality it's only one.
Then in _Content_, you'll see them separated.

Maybe I'm not explaining very well, let me know

**Comment by lppedd** [comment]
Ohhhhh wait, now I got what you meant.

So basically we'll be in a situation such that the Composite Repository will me a new tree node.
Then inside the node all the real repositories will appear.
Is that what you meant?

**Comment by JaneWardSandy** [comment]
Yes, as you said, this design is more reasonable and more user-friendly.
It's just that the scalability of my code writing is not good before, and it is not easy to increase the tree level. I may not consider providing this modification in a short time.

**Comment by JaneWardSandy** [comment]
Whether it's code cleanup or architecture, or whatever, your help would be greatly appreciated.

The original plan was to think about version 2.0 and complete the refactoring of the entire project, but currently limited by the cognition and development ability of software engineering, I have not been able to start preliminary preparations.
